### PR TITLE
Makefile refactoring.

### PR DESCRIPTION
**Pull request by sunfishcode**
These changes are factored out of #89. NFC.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -76,67 +77,80 @@ build/compiler-rt.BUILT: build/llvm.BUILT
...
-		-DLIBCXX_LIBDIR_SUFFIX=/wasm32-wasi \
-		-DWASI_SDK_PREFIX=$(PREFIX) \
-		-DCMAKE_C_FLAGS="$(DEBUG_PREFIX_MAP)" \
-		-DCMAKE_CXX_FLAGS="$(DEBUG_PREFIX_MAP)" \
-		--debug-trycompile \
-		$(LLVM_PROJ_DIR)/libcxx
+	cmake -B build/libcxx -G Ninja $(LIBCXX_CMAKE_FLAGS) \
+	    -DCMAKE_C_FLAGS="$(DEBUG_PREFIX_MAP)" \
+	    -DCMAKE_CXX_FLAGS="$(DEBUG_PREFIX_MAP)" \
+            -DLIBCXX_LIBDIR_SUFFIX=/wasm32-wasi \
```
Comment (sbc100) [inline Makefile]: indentation

File: Makefile [hunk 2]
```diff
@@ -76,67 +77,80 @@ build/compiler-rt.BUILT: build/llvm.BUILT
...
-		-DWASI_SDK_PREFIX=$(PREFIX) \
-		-DCMAKE_C_FLAGS="$(DEBUG_PREFIX_MAP) -I$(PREFIX)/share/wasi-sysroot/include" \
-		-DCMAKE_CXX_FLAGS="$(DEBUG_PREFIX_MAP) -I$(PREFIX)/share/wasi-sysroot/include/c++/v1" \
-		-DUNIX:BOOL=ON \
-		--debug-trycompile \
-		$(LLVM_PROJ_DIR)/libcxxabi
+	cmake -B build/libcxxabi -G Ninja $(LIBCXXABI_CMAKE_FLAGS) \
+	    -DCMAKE_C_FLAGS="$(DEBUG_PREFIX_MAP)" \
+	    -DCMAKE_CXX_FLAGS="$(DEBUG_PREFIX_MAP)" \
+	    -DLIBCXXABI_LIBDIR_SUFFIX=/wasm32-wasi \
```
Comment (sbc100) [inline Makefile]: Why are these three flags remaining inline?

Comment (sunfishcode) [reply Makefile]: It anticipates the LTO patch, where there are different flags for LTO and non-LTO.

File: Makefile (hunk 1 above)
Comment (sunfishcode) [reply Makefile]: My editor had inserted 8 spaces instead of a tab :-/. Fixed now.

**Pull request merged**